aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1329,7 +1329,8 @@ static Tensor select_sparse(const Tensor& self, int64_t dim, int64_t index) {
       if (new_values.size(0) == 1) {
         return new_values[0];
       } else {
-        return new_values.sum(0);
+        // sum promotes integral type to int64 when dtype is not specified.
```

Comment:
Seems like the default should be to use `self.scalar_type` but good catch!

Reply:
Yeah, it is a bit unclear, but intentional. The doc could have been on that one, for sure!